### PR TITLE
Box item spawning rework (and illegal tech introspection)

### PR DESCRIPTION
**Pull request by bloons3**
## About The Pull Request

Boxes spawn their contents with PopulateContents() when the box is created. This is fine, except no procs can _see_ into those boxes.

In order for RnD's illegal technology to see _into_ boxes in order to determine what should be illegal or not, it needs to have runtime access to box/storage contents.

## Why It's Good For The Game

image

Goal is to make this give illegal technology :)

image

## Changelog
:cl:
code: changed some code
refactor: refactored some code
/:cl:

### CODE REVIEW

**Comment by Archanial** [comment]
Can't you look into storage component of the box?

**Comment by bloons3** [comment]
> Can't you look into storage component of the box?

Some storage items call their parent (like the syndicate briefcase that generates paper/folder from the parent).

Will look into it

File: code/game/objects/items/storage/uplink_kits.dm [hunk 1]
```diff
@@ -312,6 +312,11 @@
 	desc = "A sleek, sturdy box."
 	icon_state = "syndiebox"
 	illustration = "writing_syndie"
+	var/list/items_inside = null//used for illegal tech to look inside of boxes and storage from only the typepath
+
+/obj/item/storage/box/syndie_kit/PopulateContents()
+	if(items_inside?.len)
```
Comment (Archanial) [inline code/game/objects/items/storage/uplink_kits.dm:318]:
```suggestion
	if(LAZYLEN(items_inside))
```

File: code/game/objects/items/storage/uplink_kits.dm [hunk 2]
```diff
@@ -312,6 +312,11 @@
 	desc = "A sleek, sturdy box."
 	icon_state = "syndiebox"
 	illustration = "writing_syndie"
+	var/list/items_inside = null//used for illegal tech to look inside of boxes and storage from only the typepath
```
Comment (Archanial) [inline code/game/objects/items/storage/uplink_kits.dm:315]:
```suggestion
	var/list/items_inside		//used for illegal tech to look inside of boxes and storage from only the typepath
```

File: code/game/objects/items/storage/uplink_kits.dm [hunk 3]
```diff
@@ -325,16 +330,21 @@
 	redpaper.color = "#FF0000" //Red paper, for an extra special calling card flair
 
 /obj/item/storage/box/syndie_kit/imp_freedom
+	items_inside = list(
+		/obj/item/implanter/freedom = 1
+	)
 	name = "freedom implant box"
```
Comment (Archanial) [inline code/game/objects/items/storage/uplink_kits.dm:336]:
```suggestion
	name = "freedom implant box"
	items_inside = list(
		/obj/item/implanter/freedom = 1
	)
```

**Comment by PowerfulBacon** [comment]
null the list after its been used

**Comment by PowerfulBacon** [comment]
>  This is fine, except no procs can see into those boxes.

.contents moment

**Comment by Archanial** [comment]
> 
> 
> > This is fine, except no procs can see into those boxes.
> 
> .contents moment

The issue is that you cannot see into the contents without initializing the box. One way around that would be to initialize the box while adding items to illegal tech list and then qdel them right after.

**Comment by bloons3** [comment]
> > > This is fine, except no procs can see into those boxes.
> > 
> > 
> > .contents moment
> 
> The issue is that you cannot see into the contents without initializing the box. One way around that would be to initialize the box while adding items to illegal tech list and then qdel them right after.

Some boxes have weird spawn chances for items, so I'd need the master list which is picked from.

File: code/game/objects/items/storage/uplink_kits.dm (hunk 1 above)
Comment (PowerfulBacon) [reply code/game/objects/items/storage/uplink_kits.dm:318]: if(LAZYLEN(items_inside))

**Pull request closed without merging**